src/libraries/System.Data.OleDb/src/DbBindings.cs

```diff
@@ -4,13 +4,16 @@
 
 using System.Data.Common;
 using System.Diagnostics;
+using System.Runtime.InteropServices;
 using System.Text;
 
 namespace System.Data.OleDb
 {
     internal sealed class Bindings
     {
+        private static readonly bool s_runningOnX86 = Marshal.SizeOf<IntPtr>() == 4;
```

Comment:
This should check for x86 explicitly, using RuntimeInformation.ProcessArchitecture. This can also be running on ARM.